dot/sync/disjoint_block_set.go

```diff
@@ -44,6 +53,20 @@ type pendingBlock struct {
 	header        *types.Header
 	body          *types.Body
 	justification []byte
+
+	// the time when this block should be cleared from the set.
+	// if the block is re-added to the set, this time get updated.
+	clearAt time.Time
+}
+
+func newPendingBlock(hash common.Hash, number *big.Int, header *types.Header, body *types.Body) *pendingBlock {
+	return &pendingBlock{
+		hash:    hash,
+		number:  number,
+		header:  header,
+		body:    body,
+		clearAt: time.Now().Add(ttl),
```

Comment:
Should we inject the current time? :thinking: For future testability purposes?

Reply:
updated